### PR TITLE
Reenable Recaptcha

### PR DESCRIPTION
**Pull request by wrandall22**
This PR re-enables Recaptcha, but with some extra checks to ensure we don't fail when Recaptcha returns us a message instead of data.

### CODE REVIEW

File: src/common/components/Recaptcha/Recaptcha.tsx [hunk 1]
```diff
@@ -71,50 +71,50 @@ export const Recaptcha = ({
...
+      try {
+        const token = await grecaptcha.execute(recaptchaKey, { action: action })
+        const serverResponse = await fetch(`${apiUrl}/recaptcha/verify`, {
+          method: 'POST',
+          body: JSON.stringify({ token: token }),
+          headers: { 'Content-Type': 'application/json' }
+        })
+        const data = await serverResponse.json()
+
+        if (!data || !data.score || !data.action) {
```
Comment (dr-bizz) [inline src/common/components/Recaptcha/Recaptcha.tsx:89]: Are we positive that the action was returning `undefined` and not `"undefined"` (as a string)?

**Comment by wrandall22** [comment]
Billing has already been enabled so we shouldn't hit this specific error again.

File: src/common/components/Recaptcha/Recaptcha.test.tsx [hunk 2]
```diff
@@ -238,7 +238,51 @@ describe('Recaptcha component', () => {
...
+
+    onSuccess.mockImplementation(() => console.log('success after weird'))
+
+    const { getByRole } = render(
+      buildRecaptcha()
+    )
+
+    await userEvent.click(getByRole('button'))
+    await waitFor(() => {
+      expect(onSuccess).toHaveBeenCalledTimes(1)
```
Comment (dr-bizz) [inline src/common/components/Recaptcha/Recaptcha.test.tsx:261]: You could write, but I'm not too fussy about it as this is urgent and the other tests also have this one.
```suggestion
      expect(onSuccess).toHaveBeenCalled()
```

**Review by dr-bizz — APPROVED** [review]

File: src/common/components/Recaptcha/Recaptcha.tsx (hunk 1 above)
Comment (dr-bizz) [reply src/common/components/Recaptcha/Recaptcha.tsx:89]: I see. `action` itself is undefined.

Comment (wrandall22) [reply src/common/components/Recaptcha/Recaptcha.tsx:89]: I'm about 90% sure; I think it was something like `{ message: "Over Enterprise free quota. Please ensure your project has an active billing account" }` as a response.

File: src/common/components/Recaptcha/Recaptcha.test.tsx (hunk 2 above)
Comment (wrandall22) [reply src/common/components/Recaptcha/Recaptcha.test.tsx:261]: I like having it as 1 time to ensure the return statements are in effect and not double submitting the gift.

Comment (dr-bizz) [reply src/common/components/Recaptcha/Recaptcha.test.tsx:261]: That's fine my me

**Pull request merged**